src/frontmatter/frontmatter.ts

```diff
@@ -15,11 +15,11 @@ const safeParse = (value: string) => {
 };
 
 export class Frontmatter<T extends object = any> implements IFrontmatter<T> {
-  public readonly document: Unist.Parent;
+  public readonly document: Unist.Node;
   private readonly node: Unist.Literal;
   private properties: Partial<T> | null;
 
-  constructor(data: Unist.Parent | string, mutate = false) {
+  constructor(data: Unist.Node | string, mutate = false) {
```

Comment:
It must be a parent. A regular node cannot have children, so well, you should have a condition somewhere else.

Reply:
This shouldn't be a breaking change, I account for it here: https://github.com/stoplightio/markdown/pull/12/files#diff-8a4a4f056b83cae6c03615b0469e5e55R99

Just figured might as well do this check once here rather than in every single place that wants to use the Frontmatter stuff?

Reply:
I'd throw a TypeError if a wrong node is provided, but I get your point

Reply:
Then we'd need to try/catch all over the place 🤷‍♂.